src/Temporalio/Client/WorkflowUpdateHandle.cs

```diff
@@ -0,0 +1,92 @@
+using System;
+using System.Threading.Tasks;
+using Temporalio.Api.Update.V1;
+
+namespace Temporalio.Client
+{
+    /// <summary>
+    /// Workflow update handle to perform actions on an individual workflow update.
+    /// </summary>
+    /// <param name="Client">Client used for update handle calls.</param>
+    /// <param name="Id">Update ID.</param>
+    /// <param name="WorkflowId">Workflow ID.</param>
+    /// <param name="WorkflowRunId">Run ID if any.</param>
+    /// <remarks>WARNING: Workflow update is experimental and APIs may change.</remarks>
+    public record WorkflowUpdateHandle(
+        ITemporalClient Client,
+        string Id,
+        string WorkflowId,
+        string? WorkflowRunId = null)
+    {
+        /// <summary>
+        /// Gets the known outcome.
+        /// </summary>
+        internal Outcome? KnownOutcome { private get; init; }
+
+        /// <summary>
+        /// Wait for an update result disregarding any return value.
+        /// </summary>
+        /// <param name="timeout">Optional timeout. Defaults to 1 minute, but can be set to
+        /// <see cref="System.Threading.Timeout.InfiniteTimeSpan" /> but probably shouldn't. A
+        /// timeout will throw a <see cref="OperationCanceledException" />.</param>
+        /// <param name="rpcOptions">Extra RPC options.</param>
+        /// <returns>Completed update task.</returns>
+        /// <remarks>WARNING: Workflow update is experimental and APIs may change.</remarks>
+        public Task GetResultAsync(TimeSpan? timeout = null, RpcOptions? rpcOptions = null) =>
```

Comment:
I'm a bit inclined to say this should actually have a different name if it disregards the result. Like `WaitForCompletedAsync`.

But, ignoring that - why even bother with this separately. Can't someone just use the result-returning version and just not use the result?

Reply:
> I'm a bit inclined to say this should actually have a different name if it disregards the result. Like WaitForCompletedAsync.

This matches the workflow get result (and we tend to kinda call it that for things that can optionally return results like workflows and activities). Users may get confused that the presence of a return value changes the function name since other than return value they are the same.

> But, ignoring that - why even bother with this separately. Can't someone just use the result-returning version and just not use the result?

In .NET `Task` and `Task<Result>` are distinctly defined and there is no "unit" or "void" type for these purposes in .NET. So every .NET call (e.g. [Task.Run](https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.run)) has to differentiate.

Reply:
Aaah, ok, makes sense.